src/deep_links/mod.rs

```diff
@@ -300,6 +380,23 @@ pub struct StreamDeepLinks {
     pub external_player: ExternalPlayerLink,
 }
 
+impl From<(&Stream, &Settings)> for StreamDeepLinks {
+    fn from((stream, settings): (&Stream, &Settings)) -> Self {
+        StreamDeepLinks {
+            player: stream
+                .encode()
+                .map(|stream| {
+                    format!(
+                        "stremio:///player/{}",
+                        utf8_percent_encode(&stream, URI_COMPONENT_ENCODE_SET),
+                    )
+                })
+                .unwrap_or_else(|error| ErrorLink::from(error).into()),
+            external_player: ExternalPlayerLink::from((stream, settings)),
+        }
+    }
+}
+
 impl From<(&Stream, &Option<Url>, &Settings)> for StreamDeepLinks {
```

Comment:
- [ ] Should be removed if we don't want to be able to override the `streaming_url`